File: docs/js精粹/typescript/Typescript原理与高级类型讲解 ②/ts2/1.ts
function mixin<T extends object, U extends object>(first: T, second: U): T & U {
  const result = <T & U>{};
  for (let id in first) {
    (<T>result)[id] = first[id];
  }
  for (let id in second) {
    if (!result.hasOwnProperty(id)) {
      (<U>result)[id] = second[id];
    }
  }

  return result;
}

const x = mixin({ a: 'hello' }, { b: 42 });

// 现在 x 拥有了 a 属性与 b 属性
console.log(x.a);
console.log(x.b);

File: docs/rust/demos/1.rs
// rust 入门

fn main() {
    // 必须以 ; 结尾
    println!("hello rust");
    /** 打印函数 */
    println!("{}", add(1, 2));
}

fn add(first: i32, sec: i32) -> i32 {
    return first + sec;
}
